core/src/main/kotlin/io/customer/sdk/data/store/Client.kt

```diff
@@ -1,83 +1,45 @@
-
-    /**
-     * Simpler class for ReactNative clients.
-     */
-    class ReactNative(sdkVersion: String) : Client(
-        source = SOURCE_REACT_NATIVE,
-        sdkVersion = sdkVersion
-    )
-
-    /**
-     * Simpler class for Expo clients.
-     */
-    class Expo(sdkVersion: String) : Client(source = SOURCE_EXPO, sdkVersion = sdkVersion)
-
-    /**
-     * Simpler class for Flutter clients.
-     */
-    class Flutter(sdkVersion: String) : Client(source = SOURCE_FLUTTER, sdkVersion = sdkVersion)
-
-    /**
-     * Other class to allow adding custom sources for clients that are not
-     * supported above.
-     * <p/>
-     * Use this only if the client platform is not available in the above list.
-     */
-    class Other internal constructor(
-        source: String,
-        sdkVersion: String
-    ) : Client(source = source, sdkVersion = sdkVersion)
-
     companion object {
-        internal const val SOURCE_ANDROID = "Android"
-        internal const val SOURCE_REACT_NATIVE = "ReactNative"
-        internal const val SOURCE_EXPO = "Expo"
-        internal const val SOURCE_FLUTTER = "Flutter"
+        private const val SOURCE_ANDROID = "Android"
+        internal const val META_DATA_USER_AGENT = "io.customer.sdk.android.core.USER_AGENT"
+        internal const val META_DATA_SDK_VERSION = "io.customer.sdk.android.core.SDK_VERSION"
 
         /**
-         * Helper method to create client from raw values
+         * Creates a new [Client] instance from the manifest meta-data.
+         * If the user-agent or SDK version is not found, the default client is returned.
+         * Default client is created with [SOURCE_ANDROID] and SDK version mentioned in [Version] class.
          *
-         * @param source raw string of client source (case insensitive)
-         * @param sdkVersion version of the SDK used
-         * @return [Client] created from provided values
+         * @param metadata Android application meta-data to retrieve the user-agent and SDK version from.
+         * @return The client instance created from the manifest meta-data.
+         * If not found, the default client is returned.
          */
-        fun fromRawValue(source: String, sdkVersion: String): Client = when {
-            source.equals(
-                other = SOURCE_ANDROID,
-                ignoreCase = true
-            ) -> Android(sdkVersion = sdkVersion)
-            source.equals(
-                other = SOURCE_REACT_NATIVE,
-                ignoreCase = true
-            ) -> ReactNative(sdkVersion = sdkVersion)
-            source.equals(
-                other = SOURCE_EXPO,
-                ignoreCase = true
-            ) -> Expo(sdkVersion = sdkVersion)
-            source.equals(
-                other = SOURCE_FLUTTER,
-                ignoreCase = true
-            ) -> Flutter(sdkVersion = sdkVersion)
-            else -> Other(source = source, sdkVersion = sdkVersion)
+        fun fromMetadata(metadata: Bundle?): Client {
```

Comment:
what would happen if both android SDK and react native SDK manifest have same values? shouldn't we override or handle that case?

Reply:
Added values in common test to help verify manifest merger challenges